gems/smithy/lib/smithy/plan.rb

```diff
@@ -9,9 +9,17 @@ class Plan
     def initialize(model, type, options = {})
       @model = model
       @type = type
-      @options = options
       @service = find_service(model['shapes'])
 
+      @name = options.fetch(:name, default_name(service))
+      @identifier = @name.downcase
+      @module_name = options.fetch(:module_name, @name)
+      @gem_name = options.fetch(:gem_name, default_gem_name(@module_name, @type))
+      @gem_version = options.fetch(:gem_version)
```

Comment:
Since there's no default value set on `:gem_version`, this will raise `KeyError` - should there be a default in place to prevent this from happening? The initialize params does not indicate that this is a required in the options hash

Reply:
I think we should explicitly require gem_version - this prevents overwrites of existing versions.